microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/utils/DependencyUtil.java

```diff
@@ -72,4 +90,39 @@ private static File getLocalArtifactFile(ArtifactKey a) {
 		}
 		return null;
 	}
+
+	public static List<com.redhat.microprofile.jdt.core.ArtifactResolver.Artifact> getDependencies(String groupId,
+			String artifactId, String version, IProgressMonitor monitor) throws CoreException {
+		org.eclipse.aether.artifact.Artifact artifact = new DefaultArtifact(groupId, artifactId, null, version);
+		ICallable<ArtifactDescriptorResult> callable = new ICallable<ArtifactDescriptorResult>() {
+			public ArtifactDescriptorResult call(IMavenExecutionContext context, IProgressMonitor monitor)
+					throws CoreException {
+				return resolveDescriptor(artifact, context.getRepositorySession());
+			}
+		};
+
+		ArtifactDescriptorResult result = MavenPlugin.getMaven().execute(callable, monitor);
+		if (result != null) {
+			return result.getDependencies().stream()
+					.map(dep -> new com.redhat.microprofile.jdt.core.ArtifactResolver.Artifact(
+							dep.getArtifact().getGroupId(), dep.getArtifact().getArtifactId(),
+							dep.getArtifact().getVersion(), dep.getArtifact().getClassifier()))
+					.collect(Collectors.toList());
+		}
+		return Collections.emptyList();
+	}
+
+	private static ArtifactDescriptorResult resolveDescriptor(org.eclipse.aether.artifact.Artifact artifact,
+			RepositorySystemSession repoSession) {
+		try {
+			RepositorySystem repoSystem = MavenPluginActivator.getDefault().getRepositorySystem();
+			repoSession = new DefaultRepositorySystemSession(repoSession);
+			return repoSystem.readArtifactDescriptor(repoSession,
+					new ArtifactDescriptorRequest().setArtifact(artifact));
+		} catch (ArtifactDescriptorException e) {

```

Comment:
please add some logging

Reply:
@fbricon I throw the error now instead of catching it.